tests/testtopic/test_testtopic.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+from faust.windows import TumblingWindow
+
+from kafkaaggregator.testtopic.agents import count, process_stream
+from kafkaaggregator.testtopic.models import TestTopic
+
+
+@pytest.mark.asyncio
+async def test_tumbling_window_ranges():
+    window_size = 1.0
+    window = TumblingWindow(window_size)
+    # Test if a timestamp in the extremes of the window falls in the expected
+    # window
+    start = 0
+    end = 0.9
+    assert window.ranges(start) == [(start, end)]
+    assert window.ranges(end) == [(start, end)]
+    assert window.ranges(start + 1.0) == [(start + 1.0, end + 1.0)]
+    # Test if (timestamp > 0.9) and  (timestamp < 1.0) falls in the expected
+    # window
+    assert window.ranges(start + 0.99) == [(start, end)]
+
+
+@pytest.mark.asyncio
+async def test_count_table(test_app):
+    """Test if the count table is updated every time the agent processes a new
+    message
+    """
+    async with process_stream.test_context() as agent:
+        test_topic_1 = TestTopic(time=0.0, value=0.5)
+        await agent.put(test_topic_1)
+        assert count["test_topic"] == 1
+        await agent.put(test_topic_1)
+        assert count["test_topic"] == 2
+
+
+@pytest.mark.asyncio
+async def test_count_table_persistence(test_app):
```

Comment:
Just a note that this test could be a bit fragile — I think that if any test uses the `test_app` fixture after `test_count_table` but before this one, then the test will fail. An equivalent way to demonstrate the same thing might be to put two `async with process_stream.test_context() as agent:` lines in the same test function and test for `count` and `count + 1`.

Reply:
Thanks, will fix that as I add more tests.